src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -194,11 +194,15 @@ def execute(self, args):
         import knack.events as events
         from knack.util import CommandResultItem, todict
 
+        events.EVENT_INVOKER_PRE_TRUNCATE_CMD_TBL = 'CommandInvoker.OnPreTruncateCommandTable'
```

Comment:
Do we have a centralized place to put the event definition? This shouldn't be littered around.

Reply:
I don't think there is a centralized place for knack event definitions in the core, they are defined in the [knack repo](https://github.com/Microsoft/knack/blob/master/knack/events.py). I thought we were planning to put the new event in the core first before moving it to knack

Reply:
@cheeong Yes you're correct but there's no need to add a new property to `knack.events....`.
You should be able to just define your own string and raise that event directly.

When/if we do move it to knack, then you can use it from there.
Does that make sense?

Reply:
I see, thanks!